### PR TITLE
use protobuf when possible

### PR DESCRIPTION
**Pull request by deads2k**
/hold

holding until we prove it works.

### CODE REVIEW

**Comment by deads2k** [comment]
@smarterclayton spelled like this?

**Comment by enj** [comment]
Does this work with the dynamic client?

File: pkg/config/client/client_config.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ func applyClientConnectionOverrides(overrides *ClientConnectionOverrides, kubeCo
 	if len(overrides.ContentType) > 0 {
 		kubeConfig.ContentConfig.ContentType = overrides.ContentType
 	}
+
+	// if we have no preferences at this point, claim that we accept both proto and json.  We will get proto if the server supports it.
+	// this is a slightly niggly thing.  If the server has proto and our client does not (possible, but not super likely) then this fails.
+	if len(kubeConfig.ContentConfig.AcceptContentTypes) == 0 {
+		kubeConfig.ContentConfig.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
+	}
+	if len(kubeConfig.ContentConfig.AcceptContentTypes) == 0 {
```
Comment (smarterclayton) [inline pkg/config/client/client_config.go]: This is duplication?

**Comment by smarterclayton** [comment]
We need the dynamic client to bypass proto somehow.  If you are using unstructured you require json.  I can look at sollys watch bug (that should be easy to review), but we may want to make dynamic client clearly override / default to json even if we want to leave open that we may have a more efficient unstructured payload in the future.

**Pull request merged**